types/express-serve-static-core/index.d.ts

```diff
@@ -292,12 +292,22 @@ export interface IRouter extends RequestHandler {
     /**
      * Stack of configured routes
      */
-    stack: any[];
+    stack: ILayer[];
+}
+
+export interface ILayer {
+    route: IRoute;
```

Comment:
`route` can be `undefined` 

https://github.com/expressjs/express/blob/4b9cd2fd0e13519a16bc36e8b4212e7924698b2e/lib/router/index.js#L481

```suggestion
    route?: IRoute;
```